src/lib/snarky_pairing/g2_precomputation.ml

```diff
@@ -36,22 +34,90 @@ struct
   open Impl
   open Let_syntax
 
-  type loop_state = {rx: Fqe.t; ry: Fqe.t}
+  type 'fqe loop_state = {rx: 'fqe; ry: 'fqe}
 
   let length (a, b, c) =
     let l = Field.Checked.length in
     max (l a) (max (l b) (l c))
 
+  let doubling_step_unchecked (s : Fqe.Unchecked.t loop_state) =
```

Comment:
can you use the SNARK checked/unchecked quickcheck for doubling_step and addition_step?

Reply:
can this happen later? I have everything working and know that this is correct

Reply:
:+1: